ext/soap/php_encoding.c

```diff
@@ -1380,8 +1391,15 @@ static zval *to_zval_object_ex(zval *ret, encodeTypePtr type, xmlNodePtr data, z
 	} else if (SOAP_GLOBAL(class_map) && type->type_str) {
 		zval              *classname;
 		zend_class_entry  *tmp;
-
-		if ((classname = zend_hash_str_find_deref(SOAP_GLOBAL(class_map), type->type_str, strlen(type->type_str))) != NULL &&
+        classname = zend_hash_str_find_deref(SOAP_GLOBAL(class_map), type->type_str, strlen(type->type_str));
+        if(classname == NULL){
+            if (type->ns) {
+                zend_string *nscat =zend_strpprintf(0, "{%s}%s", type->ns, type->type_str);
+                classname = zend_hash_find_deref(SOAP_GLOBAL(class_map),nscat);
```

Comment:
- I think it's better to change the order (first search in namespace). 
- Usage of ``zend_strpprintf()`` on each lookup looks inefficient (additional memory allocation/deallocation). It would be great to find a better solution.